File: 0495-Teemo Attacking/0495-Teemo Attacking.cpp
class Solution {
public:
    int findPoisonedDuration(vector<int>& timeSeries, int duration) {
        int n = timeSeries.size();
        if (n == 0)
        {
            return 0;
        }

        int count = 0;
        
        for (int i = 1; i < n; ++i)
        {
            if (timeSeries[i] < timeSeries[i - 1] + duration)
            {
                count += timeSeries[i] - timeSeries[i - 1];
            }
            else
            {
                count += duration;
            }
        }
        
        count += duration;
        
        return count;
    }
};

File: 0401-0500/0495-Teemo Attacking/0495-Teemo Attacking.py
class Solution:
    def findPoisonedDuration(self, timeSeries: List[int], duration: int) -> int:
        if not timeSeries:
            return 0
        total = duration
        for i in range(1, len(timeSeries)):
            total += min(duration, timeSeries[i] - timeSeries[i - 1])
        return total
